File: public/usage-examples/generative_ai/generate-convo-1-example.cpp
#include "splashkit.h"
#include <string>
#include <iostream>

int main()
{
    // Initialize SplashKit
    open_window("AI Conversation Example", 800, 600);
    
    // Display welcome message
    write_line("=== Terminal AI Conversation ===");
    write_line("Type your message and press Enter to chat with AI");
    write_line("Type 'quit' to exit");
    write_line("");
    
    std::string user_message;
    
    // Simple conversation loop
    while (true)
    {
        // Get user input
        write("You: ");
        user_message = read_line();
        
        // Check if user wants to quit
        if (user_message == "quit")
        {
            write_line("Goodbye!");
            break;
        }
        
        // Skip empty messages
        if (user_message.empty())
        {
            continue;
        }
        
        // Generate AI response using generate_text()
        std::string ai_response = generate_text(user_message);
        
        // Display AI response
        write_line("AI: " + ai_response);
        write_line("");
    }
    
    close_window("AI Conversation Example");
    return 0;
}

File: public/usage-examples/generative_ai/generate-convo-1-example.py
# Simple Terminal AI Conversation - Python Example

from splashkit import *

def main():
    """Main function to run the AI conversation"""
    
    # Display welcome message
    clear_screen()
    print("=== Terminal AI Conversation ===")
    print("Type your message and press Enter to chat with AI")
    print("Type 'quit' to exit")
    print("")
    
    # Simple conversation loop
    while True:
        # Get user input
        user_message = input("You: ")
        
        # Check if user wants to quit
        if user_message.lower() == "quit":
            print("Goodbye!")
            break
        
        # Skip empty messages
        if not user_message.strip():
            continue
        
        # Generate AI response using generate_text()
        ai_response = generate_text(user_message)
        
        # Display AI response
        print(f"AI: {ai_response}")
        print("")

# Run the program
if __name__ == "__main__":
    main()
